mungegithub/mungers/approval-handler.go

```diff
@@ -90,58 +97,162 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 		return
 	}
 
-	approverSet := sets.String{}
+	ownersMap := h.getApprovedOwners(files, createApproverSet(comments))
 
-	// from oldest to latest
-	for i := len(comments) - 1; i >= 0; i-- {
-		c := comments[i]
+	if err := h.updateNotification(obj, ownersMap); err != nil {
+		return
+	}
 
-		if !mungerutil.IsValidUser(c.User) {
-			continue
+	for _, approverSet := range ownersMap {
+		if approverSet.Len() == 0 {
+			if obj.HasLabel(approvedLabel) {
+				obj.RemoveLabel(approvedLabel)
+			}
+			return
 		}
+	}
 
-		fields := strings.Fields(strings.TrimSpace(*c.Body))
+	if !obj.HasLabel(approvedLabel) {
+		obj.AddLabel(approvedLabel)
+	}
+}
 
-		if len(fields) == 1 && strings.ToLower(fields[0]) == "/approve" {
-			approverSet.Insert(*c.User.Login)
-			continue
-		}
+func (h *ApprovalHandler) updateNotification(obj *github.MungeObject, ownersMap map[string]sets.String) error {
+	notificationMatcher := c.MungerNotificationName(approvalNotificationName)
+	comments, err := obj.ListComments()
+	if err != nil {
+		glog.Error("Could not list the comments for PR%v", obj.Issue.Number)
+		return err
+	}
+	notifications := c.FilterComments(comments, notificationMatcher)
+	var latestNotification *githubapi.IssueComment
+	if notifications == nil {
+		latestNotification = nil
```

Comment:
latestNotification will initialize to nil. This case can be deleted and only the inverse needs to be handled.